EarlGrey/Matcher/GREYMatchers.m

```diff
@@ -262,7 +262,19 @@ @implementation GREYMatchers
   return [[GREYElementMatcherBlock alloc] initWithMatchesBlock:matches descriptionBlock:describe];
 }
 
-+ (id<GREYMatcher>)matcherForProgress:(id)comparisonMatcher {
++ (id<GREYMatcher>)matcherForKindOfClassNamed:(NSString *)className {
+  NSParameterAssert(className);
+
+  MatchesBlock matches = ^BOOL(id element) {
+    return [element isKindOfClass:NSClassFromString(className)];
```

Comment:
Move NSClassFromString(className) to outside the block so it only happens once when the matcher is created and reused every time matched.